samza-core/src/main/java/org/apache/samza/processor/SamzaContainerController.java

```diff
@@ -105,7 +106,9 @@ public void startContainer(ContainerModel containerModel, Config config, int max
         localityManager,
         new JmxServer(),
         Util.<String, MetricsReporter>javaMapAsScalaMap(metricsReporterMap),
-        taskFactory);
+        taskFactory,
+        // TODO: need to use the correct local ApplicationRunner here
+        ApplicationRunner.getLocalRunner(config));
```

Comment:
Actually, this should be neither the local runner nor remote runner. It needs to be ContainerLocalRunner, which is just play a role of stream provider as we discussed today. Let me set it to null for now.